### DISCUSSION TITLE
AWS  configured properly

### DISCUSSION THREAD

**Discussion by mickymkumar** [magma/magma#6693] [Q&A]
I am getting the following error - I have my AWS configured properly

ubuntu@ip-172-31-45-146:~/magma/experimental/cloudstrapper/playbooks$ ansible-playbook aws-prerequisites.yaml -e 'awsTargetRegion=ca-central-1' -e "dirInventory=${VAR_DIR}" --tags keyCreate
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

PLAY [localhost] ****************************************************************************************************************

TASK [Gathering Facts] **********************************************************************************************************
fatal: [localhost]: FAILED! => {"msg": "The field 'environment' has an invalid value, which includes an undefined variable. The error was: 'awsAccessKey' is undefined"}

PLAY RECAP **********************************************************************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0

**Comment by arunuke** [comment]
@Micky-Kumar To deploy any resource on AWS via APIs, you would need an AWS Access Key and a Secret Key, both of which you can generate for an IAM user. The Cloudstrapper uses these keys to authenticate itself and then deploys the resources as desired. On top of the [README](https://github.com/magma/magma/tree/master/experimental/cloudstrapper) file, the prerequisites section talks about artifacts that are required before starting the installation. 

In this specific case the secrets.yaml file, which holds authentication information for AWS, Docker and Github is missing or it is incorrect.

**Reply by mickymkumar** [reply]
@arunuke All authentication information for AWS, Docker and Github is correct
and I still receiving this error

**Reply by arunuke** [reply]
Thanks @Micky-Kumar . Can you try the same command with dirLocalInventory (instead of dirInventory)? I looked at the code and this could be documentation bug that needs to be fixed.

**Reply by mickymkumar** [reply]
~/magma/experimental/cloudstrapper/playbooks$ ansible-playbook aws-prerequisites.yaml -e 'awsTargetRegion=ca-central-1' -e "dirLocalInventory=${VAR_DIR}" --tags keyCreate
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not
match 'all'
ERROR! vars file {{ dirLocalInventory }}/secrets.yaml was not found
Could not find file on the Ansible Controller.
If you are using a module and expect the file to exist on the remote, see the remote_src option

**Reply by arunuke** [reply]
Thanks @Micky-Kumar . I can only guess if there is an issue with the environment variable that is being passed. Can I request you to pass an actual directory (absolute path, not relative path), an 'ls -l' on the directory and a 'cat secrets.yaml' on the *redacted* file so that I can check for the parameters? I am currently testing this for 1.5 and it is working out of the box for me, so would be glad to understand what is missing here.

**Reply by mickymkumar** [reply]
~/magma/experimental/cloudstrapper/playbooks$ ls -l
total 56
-rw-rw-r--  1 ubuntu ubuntu  402 Apr 29 20:52 agw-configure.yaml
-rw-rw-r--  1 ubuntu ubuntu  510 Apr 29 20:52 agw-provision.yaml
-rw-rw-r--  1 ubuntu ubuntu  254 Apr 29 20:52 ami-configure.yaml
-rw-rw-r--  1 ubuntu ubuntu  347 Apr 29 20:52 ami-init.yaml
-rw-rw-r--  1 ubuntu ubuntu  354 Apr 29 20:52 aws-prerequisites.yaml
-rw-rw-r--  1 ubuntu ubuntu  657 Apr 29 20:52 build-configure.yaml
-rw-rw-r--  1 ubuntu ubuntu  418 Apr 29 20:52 build-provision.yaml
-rw-rw-r--  1 ubuntu ubuntu  276 Apr 29 20:52 cleanup.yaml
-rw-rw-r--  1 ubuntu ubuntu  359 Apr 29 20:52 cloudstrapper-provision.yaml
-rw-rw-r--  1 ubuntu ubuntu  424 Apr 29 20:52 devops-configure.yaml
-rw-rw-r--  1 ubuntu ubuntu  355 Apr 29 20:52 devops-init.yaml
-rw-rw-r--  1 ubuntu ubuntu  460 Apr 29 20:52 devops-provision.yaml
-rw-rw-r--  1 ubuntu ubuntu  384 Apr 29 20:52 orc8r.yaml
drwxrwxr-x 22 ubuntu ubuntu 4096 Apr 29 20:52 roles
-rw-rw-r--  1 ubuntu ubuntu    0 Apr 29 21:33 {msg:
ubuntu@ip-172-31-45-146:~/magma/experimental/cloudstrapper/playbooks$ cat secrets.yaml
cat: secrets.yaml: No such file or directory

**Reply by arunuke** [reply]
Thanks @Micky-Kumar. The error occurs because there is no secrets.yaml file in the location you are passing in the command line. 


As described in the prerequisites section, the secrets.yaml file needs to be present in a directory *outside*of the github folders. If you have not created a secrets.yaml file with your credentials yet, there is a [sample file](https://github.com/magma/magma/blob/master/experimental/cloudstrapper/playbooks/roles/vars/secrets.yaml) that you can use to populate the local copy. As described in the prerequisites section, please pass the location of the folder hosting the file for dirLocalInventory (directory on the local host that holds inventory - hence the naming).

**Reply by tashingamusiya** [reply]
Hi @arunuke @micky-kumar was this resolved I 
![ok](https://user-images.githubusercontent.com/22387604/146054447-3b4ec150-22dc-45ff-8ff5-212821672609.jpg)
have been having the exact problem for sometime.

**Reply by tashingamusiya** [reply]
@arunuke please assist if you can thank you.
![ok](https://user-images.githubusercontent.com/22387604/146057755-1de11e84-8314-4791-af3c-94c082727b74.jpg)

**Reply by tashingamusiya** [reply]
![Uploading ok2.jpg…]()

**Reply by tashingamusiya** [reply]
![ok2](https://user-images.githubusercontent.com/22387604/146160860-726ce9e8-afd7-4b42-ac6f-85b9f5f8cc31.jpg)

**Reply by mickymkumar** [reply]
was this resolved I ![ok](https://user-images.githubusercontent.com/22387604/146054447-3b4ec150-22dc-45ff-8ff5-212821672609.jpg) have been having the exact problem for sometime.

**Yes see above comments**

**Reply by tashingamusiya** [reply]
thanks for the reply much appreciated, so maybe it's a different problem I am having because you will see in the screenshot above I show that the secrets file is present but somehow it keeps mising it even after putting eaxct path something not right either syntax would you have the correct command ?

**Reply by tashingamusiya** [reply]
@arunuke @micky-kumar

**Comment by arunuke** [comment]
Hello @tashingamusiya 

Did you get a chance to look at the documentation available from [Magma Docs](https://magma.github.io/magma/docs/next/general/aws_cloudstrapper) that describes this process? From what I see, it appears you are setting secrets.yaml in the *repo* instead of your configuration environment. From the documentation I cited above, 

_Copy the secrets.yaml file and update the credentials for AWS:

cp roles/vars/secrets.yaml ~/magma-dev/_
